pkg/ledger/ledger.go

```diff
@@ -88,12 +87,30 @@ func (l *Ledger) Commit(ctx context.Context, txsData []core.TransactionData) (*C
 		}
 	}
 
-	l.monitor.CommittedTransactions(ctx, l.store.Name(), result)
-	return result, nil
+	for _, t := range txsData {
+		if accMeta, ok := t.Metadata["set_account_meta"]; ok {
+			for addr, m := range accMeta.(map[string]any) {
+				if err := l.store.UpdateAccountMetadata(ctx,
+					addr, m.(map[string]any), time.Now().Round(time.Second).UTC()); err != nil {
+					return nil, err
+				}
+			}
+		}
+	}
+
+	l.monitor.CommittedTransactions(ctx, l.store.Name(), commitRes)
+	for _, t := range txsData {
```

Comment:
Maybe we should add account metadata to transaction commit event.
Wdyt? (not for this PR but we can plan the change).